### DISCUSSION TITLE
What is the difference between these two ways to get the triggering commit?

### DISCUSSION THREAD

**Discussion by roddds** [community/community#25318] [Actions]
<p>I need to run a workflow when a Pull Request closes, and it depends on the SHA of the most recent commit before the pull request was closed. However, when looking into ways to do that, I’ve found that <code>$GITHUB_SHA</code> did not, in fact, have that - instead, I could only find it by grabbing it from the <code>github</code> context, in the path <code>github.event.pull_request.head.sha</code>.</p>
<p>These are my results from running these two commands in my action:</p>
<pre><code class="lang-auto">      - name: sha from context
        run: echo ${{ github.event.pull_request.head.sha }}

      - name: sha from env var
        run: echo $GITHUB_SHA
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5a9ac04d3783f431c26d385ffd57a3e35f3b68fc.png" data-download-href="/uploads/short-url/cVwwyUpt6QdhVI1bWB1Qw3hVETy.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/516447/181090416-0bbe48f1-f944-454f-8522-6e4e6f2b59b7.png" alt="image" data-base62-sha1="cVwwyUpt6QdhVI1bWB1Qw3hVETy" width="690" height="332" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/5a9ac04d3783f431c26d385ffd57a3e35f3b68fc_2_690x332.png, https://aws1.discourse-cdn.com/github/original/2X/5/5a9ac04d3783f431c26d385ffd57a3e35f3b68fc.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/5/5a9ac04d3783f431c26d385ffd57a3e35f3b68fc.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5a9ac04d3783f431c26d385ffd57a3e35f3b68fc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">851×410 35.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>As you can see, what I get from <code>$GITHUB_SHA</code> does not match the hash displayed under the commit message. Why are these two different?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>The GITHUB_SHA of <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request" rel="nofollow noopener">pull_request event</a> is the Last merge commit on the <code>GITHUB_REF</code> branch . The <code>GITHUB_REF</code> is PR merge branch <code>refs/pull/:prNumber/merge</code> .<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b1785b2bd71c7d8ae55b8440e2e42ecde4d83e25.png" data-download-href="/uploads/short-url/pjYnc3KZ9mflWNeYT48BXpOQwsZ.png?dl=1" title="PR SHA" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181090418-9489bc73-b3bd-4483-9d70-06281b0be74f.png" alt="PR SHA" data-base62-sha1="pjYnc3KZ9mflWNeYT48BXpOQwsZ" width="577" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b1785b2bd71c7d8ae55b8440e2e42ecde4d83e25_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">PR SHA</span><span class="informations">776×672 25 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Pull request head points to the source branch of the pull request.<br>
<code>github.event.pull_request.head.sha</code> is the right way to get SHA of the most recent commit before the pull request was closed.</p>
